src/osd/PG.cc

```diff
@@ -2101,10 +2100,12 @@ void PG::repair_object(
 void PG::forward_scrub_event(ScrubAPI fn, epoch_t epoch_queued, std::string_view desc)
 {
   dout(20) << __func__ << ": " << desc << " queued at: " << epoch_queued << dendl;
-  if (is_active() && m_scrubber) {
+  ceph_assert(m_scrubber);
+  if (is_active()) {
     ((*m_scrubber).*fn)(epoch_queued);
   } else {
     // pg might be in the process of being deleted
+    m_scrubber->clear_queued_or_active();
```

Comment:
Why does this cause a state change in the scrubber?

Reply:
fixed